### PR TITLE
Improve portfolio EVM tests

### PR DESCRIPTION
**Pull request by mhofman**
## Summary by CodeRabbit

* **Tests**
  * Expanded test suite and added scaffolding/utilities for richer assertions and call-logging.
  * Broadened coverage for deposit, withdraw, rebalance flows, including many edge cases (validation, unknown chains, allocation scenarios) and flow tracking.
  * Public API and production behavior unchanged.

**Note:** Internal quality improvements only; no user-facing changes.

### CODE REVIEW

File: packages/portfolio-contract/test/portfolio.exo.test.ts [hunk 1]
```diff
@@ -1,30 +1,66 @@
 /** @file tests for PortfolioKit exo */
+/* eslint-disable no-sparse-arrays */
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/portfolio-contract/test/portfolio.exo.test.ts:2]: The suggested change is intractable for this test, and test code doesn't need to be as tight as the actual sources.

File: packages/portfolio-contract/test/portfolio.exo.test.ts [hunk 2]
```diff
@@ -1,30 +1,66 @@
...
+    Object.entries(stubs).map(([k, stub]) => [
+      k,
+      (...args) => {
+        log.push([k, ...args]);
+        return stub(...args);
+      },
+    ]),
+  );
+
+  // @ts-expect-error generics
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/portfolio-contract/test/portfolio.exo.test.ts]: Done.

File: packages/portfolio-contract/test/portfolio.exo.test.ts [hunk 3]
```diff
@@ -1,30 +1,66 @@
...
 const makeTestSetup = () => {
   const zone = makeHeapZone();
   const board = makeFakeBoard();
   const marshaller = board.getReadonlyMarshaller();
   const vowTools = prepareVowTools(zone);
 
+  const depStubs = {
+    rebalance: () => vowTools.asVow(() => {}),
+    executePlan: () => vowTools.asVow(() => {}),
+  } as Pick<PortfolioKitDeps, 'rebalance' | 'executePlan'>;
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/portfolio-contract/test/portfolio.exo.test.ts]: Thanks, got it.

**Review by michaelfig — APPROVED** [review]

**Comment by michaelfig** [comment]
@Mergifyio refresh
